### PR TITLE
Feature/dx 475 swish redirect

### PR DESCRIPTION
**Pull request by marinafrydenb**
Made some fixed to Swish Redirect. Its not quite done. But i would love to have some feedback to have a more clear sense of whats missing

### CODE REVIEW

File: payments/swish/redirect.md [hunk 1]
```diff
@@ -119,139 +92,161 @@ sequenceDiagram
...
-A reversal transcation need to match the Payee reference of a completed
-sales transaction.
-
-```mermaid
-sequenceDiagram
-  activate Merchant
-  Merchant->>-SwedbankPay: POST <Swish reversal>
-  activate SwedbankPay
-  SwedbankPay-->>-Merchant: transaction resource
+The API requests are displayed in the [purchase flow](#purchase-flow).
```
Comment (NullableInt) [inline payments/swish/redirect.md]: Use reference style links. (For consistency.)

File: payments/swish/redirect.md [hunk 2]
```diff
@@ -119,139 +92,161 @@ sequenceDiagram
...
+        "payeeInfo": {
+            "payeeId": "bbb33dc5-f44e-4af6-afc0-27fb5fa2f63a",
+            "payeeReference": "ref-123456",
+            "payeeName": "Merchant1",
+            "productCategory": "A123",
+            "orderReference": "or-123456",
+            "subsite": "MySubsite"
+        },
+        "prefillInfo": {
+            "msisdn": "+46xxxxxxxxx"
```
Comment (NullableInt) [inline payments/swish/redirect.md]:
```suggestion
            "msisdn": "+46739000001"
```
Use test data from our own resources 😄

File: payments/swish/redirect.md [hunk 3]
```diff
@@ -119,139 +92,161 @@ sequenceDiagram
...
+|  ✔︎︎︎︎︎  | └➔&nbsp;`language`                    | `string`      | nb-NO, sv-SE or en-US.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
```
Comment (NullableInt) [inline payments/swish/redirect.md]: Might want to double check that Swish supports other languages than `sv-SE` 🛌 😴 🔥 🔢

File: payments/vipps/redirect.md [hunk 4]
```diff
@@ -1,28 +1,10 @@
----
```
Comment (NullableInt) [inline payments/vipps/redirect.md]: Oh no! Where did the menu go?

File: payments/vipps/redirect.md [hunk 5]
```diff
@@ -32,24 +14,14 @@ sidebar:
   You either receive a Redirect URL to a hosted page or a JavaScript source
   in response.
 * You need to [redirect][reference-redirect] the payer to the Redirect payment
-  page or embed the script source on you site to create a
-  [Hosted View][hosted-view] in an `iFrame`; where she is prompted to enter the
-  registered mobile number.
-  This triggers a `POST` towards PayEx.
+  where the user is prompted to enter the registered mobile number.
+  This triggers a `POST` towards Swedbank Pay.
```
Comment (NullableInt) [inline payments/vipps/redirect.md]:
```suggestion
  This triggers a request towards Swedbank Pay.
```

File: payments/vipps/redirect.md [hunk 6]
```diff
@@ -58,32 +30,28 @@ consumers mobile number.
...
 
-#### General
-
-**Defining CallbackURL**: When implementing a scenario, it is optional to set
-a [`CallbackURL`][callbackurl] in the `POST` request.
-If callbackURL is set Swedbank Pay will send a postback request to this URL
-when the consumer has fulfilled the payment.
+{% include alert.html type="success" icon="link" body="**Defining
+`callbackUrl`**: When implementing a scenario, it is optional to set a
+`callbackUrl` in the `POST` request. If `callbackUrl` is set, Swedbank Pay will
```
Comment (NullableInt) [inline payments/vipps/redirect.md]:
```suggestion
`callbackUrl` in the request. If `callbackUrl` is set, Swedbank Pay will
```

File: payments/vipps/redirect.md [hunk 7]
```diff
@@ -97,46 +65,257 @@ sequenceDiagram
...
+|  ✔︎︎︎︎︎  | └➔&nbsp;`language`                    | `string`      | nb-NO                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (NullableInt) [inline payments/vipps/redirect.md]:
```suggestion
|  ✔︎︎︎︎︎  | └➔&nbsp;`language`                    | `string`      | `nb-NO`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```

File: payments/vipps/redirect.md [hunk 8]
```diff
@@ -97,46 +65,257 @@ sequenceDiagram
...
+```http
+
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "payment": "/psp/vipps/payments/84b9e6aa-b8f5-4e7f-fa2f-08d612f7dd5d",
+    "authorization": {
+        "vippsTransactionId": "5619328800",
+        "msisdn": "+47xxxxxxxx",
```
Comment (NullableInt) [inline payments/vipps/redirect.md]:
```suggestion
        "msisdn": "+4793000001",
```
Let's avoid xxxxx test data 😄

File: payments/vipps/redirect.md [hunk 9]
```diff
@@ -97,46 +65,257 @@ sequenceDiagram
...
+{:.code-header}
+**Response**
+
+```http
+
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "payment": "/psp/vipps/payments/84b9e6aa-b8f5-4e7f-fa2f-08d612f7dd5d",
```
Comment (NullableInt) [inline payments/vipps/redirect.md]:
```suggestion
    "payment": "/psp/vipps/payments/{{ page.paymentId }}",
```
You may want to rebase and use the new guids

File: payments/swish/redirect.md [hunk 10]
```diff
@@ -18,21 +18,19 @@ sidebar:
 
 {% include alert-development-section.md %}
 
->Swish is an one-phase payment instrument supported by the major Swedish banks.
- In the redirect e-commerce scenario, Swedbank Pay performs a payment that the
- payer confirms using her Swish mobile app.
- The consumer initiates the payment by supplying the Swish registered mobile
- number (MSISDN), connected to the Swish app.
+{% include jumbotron.html body="**Swish Payments Redirect** is where Swedbank
```
Comment (asbjornu) [inline payments/swish/redirect.md]: Please keep the sentence "Swish is an one-phase payment instrument supported by the major Swedish banks". It should also be made apparent how Redirect differs from Seamless View.

File: payments/swish/redirect.md [hunk 11]
```diff
@@ -119,139 +92,161 @@ sequenceDiagram
...
+### API Requests
+
+All valid options when posting in a payment with operation equal to `Purchase`.
+The `purchase` example shown below.
+
+{:.code-header}
+**Request**
+
+```http
+POST /psp/cswish/payments
```
Comment (asbjornu) [inline payments/swish/redirect.md]:
```suggestion
POST /psp/swish/payments
```

File: payments/swish/redirect.md [hunk 12]
```diff
@@ -119,139 +92,161 @@ sequenceDiagram
...
+Authorization: Bearer <AccessToken>
+Content-Type: application/json
+{
```
Comment (asbjornu) [inline payments/swish/redirect.md]:
```suggestion

{
```

File: payments/swish/redirect.md [hunk 13]
```diff
@@ -119,139 +92,161 @@ sequenceDiagram
...
+            "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/urls"
+        },
+        "payeeInfo": {
+            "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/payeeinfo"
+        }
+    },
+    "operations": [
+        {
+            "method": "PATCH",
+            "href": "http://localhost:18496/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```
Comment (asbjornu) [inline payments/swish/redirect.md]: No `localhost` URLs, please.

File: payments/swish/redirect.md [hunk 14]
```diff
@@ -119,139 +92,161 @@ sequenceDiagram
...
+            "rel": "update-payment-abort"
+        },
+        {
+            "method": "POST",
+            "href": "http://localhost:18496/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales",
```
Comment (asbjornu) [inline payments/swish/redirect.md]: No `localhost` URLs, please.

File: payments/vipps/redirect.md [hunk 15]
```diff
@@ -18,11 +18,11 @@ sidebar:
 
 {% include alert-development-section.md %}
 
->Vipps is a two-phase payment instrument supported by the major norwegian banks.
- In the redirect to Swedbank Pay Payments scenario,  Swedbank Pay
- receives a mobile number (MSISDN) from the payer through Payex Payments.
- Swedbank Pay performs a payment that the payer must confirm through the
- Vipps mobile app.
+{% include jumbotron.html body="**Vipps Payments** is a two-phase
```
Comment (asbjornu) [inline payments/vipps/redirect.md]: Why are you editing `payments/vipps/redirect.md` in a PR for Swish?

File: payments/swish/redirect.md [hunk 16]
```diff
@@ -18,21 +18,20 @@ sidebar:
 
 {% include alert-development-section.md %}
 
->Swish is an one-phase payment instrument supported by the major Swedish banks.
- In the redirect e-commerce scenario, Swedbank Pay performs a payment that the
- payer confirms using her Swish mobile app.
- The consumer initiates the payment by supplying the Swish registered mobile
- number (MSISDN), connected to the Swish app.
+{% include jumbotron.html body="Swish is a one-phase payment instrument supported by the
+major Swedish banks.**Swish Payments Redirect** is where Swedbank
```
Comment (asbjornu) [inline payments/swish/redirect.md]:
```suggestion
major Swedish banks. **Swish Payments Redirect** is where Swedbank
```

File: payments/vipps/redirect.md [hunk 17]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+## Authorizations
+
+The `authorizations` resource contains information about the authorization
+transactions made on a specific payment.
+
+{:.code-header}
+**Request**
+
+```http
+
```
Comment (asbjornu) [inline payments/vipps/redirect.md]: Please remove the superfluous line feeds.

```suggestion
```

File: payments/vipps/redirect.md [hunk 18]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+GET /psp/vipps/payments/84b9e6aa-b8f5-4e7f-fa2f-08d612f7dd5d/authorizations/<transactionId> HTTP/1.1
+Host: api.payex.com
+Authorization: Bearer <MerchantToken>
+
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
```

File: payments/vipps/redirect.md [hunk 19]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
```

File: payments/vipps/redirect.md [hunk 20]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+```
+
+{:.code-header}
+**Response**
+
+```http
+
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
```

File: payments/vipps/redirect.md [hunk 21]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "payment": "/psp/vipps/payments/{{ page.paymentId }}",
+    "authorization": {
+        "vippsTransactionId": "5619328800",
+        "msisdn": "+4793000001",
+        "id": "/psp/vipps/payments/84b9e6aa-b8f5-4e7f-fa2f-08d612f7dd5d/authorizations/3bfb8c66-33be-4871-465b-08d612f01a53",
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
        "id": "/psp/vipps/payments/{{ page.paymentId }}/authorizations/{{ page.transactionId }}",
```

File: payments/vipps/redirect.md [hunk 22]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        "transaction": {
+            "id": "/psp/vipps/payments/84b9e6aa-b8f5-4e7f-fa2f-08d612f7dd5d/transactions/3bfb8c66-33be-4871-465b-08d612f01a53",
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/transactions/{{ page.transactionId }}",
```

File: payments/vipps/redirect.md [hunk 23]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+{:.code-header}
+**Response**
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "payment": {
+        "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c",
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
        "id": "/psp/vipps/payments/{{ page.paymentId }}",
```

File: payments/vipps/redirect.md [hunk 24]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+            "hostUrls": "https://example.net",
+            "completeUrl": "https://example.net/payment-completed",
+            "cancelUrl": "https://example.net/payment-canceled",
+            "callbackUrl": "https://example.net/payment-callback",
+            "logoUrl": "https://example.net/payment-logo.png",
+            "termsOfServiceUrl": "https://example.net/payment-terms.pdf",
+            "paymentUrl": "https://example.net/payment-cart"
+        },
+        "payeeInfo": {
+            "payeeId": "12345678-1234-1234-1234-123456789012",
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "payeeId": "{{ page.merchantId }}",
```

File: payments/vipps/redirect.md [hunk 25]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        "remainingCaptureAmount": 1500,
+        "remainingCancellationAmount": 1500,
+        "remainingReversalAmount": 0,
+        "description": "Test Purchase",
+        "payerReference": "AB1234",
+        "initiatingSystemUserAgent": "PostmanRuntime/3.0.1",
+        "userAgent": "Mozilla/5.0...",
+        "language": "nb-NO",
+        "prices": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/prices"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/prices"
```

File: payments/vipps/redirect.md [hunk 26]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        },
+        "transactions": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/transactions"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/transactions"
```

File: payments/vipps/redirect.md [hunk 27]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        },
+        "authorizations": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/authorizations"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/authorizations"
```

File: payments/vipps/redirect.md [hunk 28]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        },
+        "reversals": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/reversals"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/reversals"
```

File: payments/vipps/redirect.md [hunk 29]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        },
+        "cancellations": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/cancellations"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/cancellations"
```

File: payments/vipps/redirect.md [hunk 30]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        },
+        "urls": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/urls"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/urls"
```

File: payments/vipps/redirect.md [hunk 31]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        },
+        "payeeInfo": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/payeeInfo"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/payeeInfo"
```

File: payments/vipps/redirect.md [hunk 32]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        },
+        "settings": {
+            "id": "/psp/vipps/payments/5adc265f-f87f-4313-577e-08d3dca1a26c/settings"
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "id": "/psp/vipps/payments/{{ page.paymentId }}/settings"
```

File: payments/vipps/redirect.md [hunk 33]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+        }
+    },
+    "operations": [
+        {
+            "method": "POST",
+            "href": "http://localhost:15486/psp/vipps/payments/c50eef6d-a788-4ec6-64ff-08d4b16740a4/authorizations",
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/vipps/payments/{{ page.paymentId }}/authorizations",
```

File: payments/vipps/redirect.md [hunk 34]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+            "rel": "create-authorization"
+        },
+        {
+            "method": "PATCH",
+            "href": "http://localhost:15486/psp/vipps/payments/c50eef6d-a788-4ec6-64ff-08d4b16740a4",
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/vipps/payments/{{ page.paymentId }}",
```

File: payments/vipps/redirect.md [hunk 35]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+            "rel": "update-payment-abort"
+        },
+        {
+            "method": "GET",
+            "href": "http://localhost:15487/vipps/payments/authorize/8fb05a835f2fc227dc7bca9abaf649b919ba8a572deb448bff543dd5806dacb7",
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
            "href": "{{ page.apiUrl }}/vipps/payments/authorize/{{ page.paymentId }}",
```

File: payments/vipps/redirect.md [hunk 36]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+|          | └─➔&nbsp;`subsite`                    | `String(40)`  | The subsite field can be used to perform split settlement on the payment. The subsites must be resolved with Swedbank Pay reconciliation before being used.                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (asbjornu) [inline payments/vipps/redirect.md]: `subsite` is not in the provided code example. Only the properties within the code example should be described in the property table. Please remove all rows describing properties that are not within the code example.

File: payments/vipps/redirect.md [hunk 37]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+transactions made on a specific payment.
+
+{:.code-header}
+**Request**
+
+```http
+
+HTTP/1.1 200 OK
+Content-Type: application/json
+GET /psp/vipps/payments/84b9e6aa-b8f5-4e7f-fa2f-08d612f7dd5d/authorizations/<transactionId> HTTP/1.1
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
GET /psp/vipps/payments/{{ page.paymentId }}/authorizations/{{ page.transactionId }} HTTP/1.1
```

File: payments/vipps/redirect.md [hunk 38]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+Host: api.payex.com
+Authorization: Bearer <MerchantToken>
```
Comment (asbjornu) [inline payments/vipps/redirect.md]:
```suggestion
Authorization: Bearer <AccessToken>
```

File: payments/vipps/redirect.md [hunk 39]
```diff
@@ -97,46 +83,257 @@ sequenceDiagram
...
+|  ✔︎︎︎︎︎  | └➔&nbsp;`authorizations.id`                           | `string`      | The relative URI of the current authorization transactions resource.                                                                                                                                                                                                                                             |
+|  ✔︎︎︎︎︎  | └➔&nbsp;`authorizations.authorizationList`                              | `array`      |  The array of authorization transaction objects.                                                                                                                                                          |
+|  ✔︎︎︎︎︎  | └➔&nbsp;`authorizations.authorizationList[]`                            | `object`      | The `authorization` transaction object described in the authorization resource below.
```
Comment (asbjornu) [inline payments/vipps/redirect.md]: @NullableInt has created a `transaction-response.md` include file you can use here. Ask him for help.

**Pull request closed without merging**